hopsanc/src/hopsanc.cpp

```diff
@@ -0,0 +1,275 @@
+        return nullptr;
+    }
+
+    //Parse variable string
+    hopsan::HString varStr(variable);
+    hopsan::HVector<hopsan::HString> splitSys = varStr.split('|');
+    hopsan::HVector<hopsan::HString> splitVar = splitSys.last().split('.');
+    splitSys.resize(splitSys.size()-1);
+    if(splitVar.size() < 3) {
+        std::cout << "Error: Component name, port name and variable name must be specified.\n";
+        size = -1;
+        return nullptr;
+    }
+
+    //Find system
+    hopsan::ComponentSystem *pSystem = spCoreComponentSystem;
+    for(size_t i=0; i<splitSys.size(); ++i) {
+        pSystem = pSystem->getSubComponentSystem(splitSys[i]);
+        if(!pSystem) {
+            std::cout << "Error: Subsystem not found: " << splitSys[i].c_str() << "\n";
+            size = -1;
+            return nullptr;
+        }
+    }
+
+    //Find component
+    hopsan::Component *pComp = pSystem->getSubComponent(splitVar[0]);
+    if(!pComp) {
+        std::cout << "Error: No such component: " << splitVar[0].c_str() << "\n";
+        size = -1;
+        return nullptr;
+    }
+
+    //Find port
+    hopsan::Port *pPort = pComp->getPort(splitVar[1]);
+    if(!pPort) {
+        std::cout << "Error: No such port: " << splitVar[1].c_str() << "\n";
+        size = -1;
+        return nullptr;
+    }
+
+    int varId = pPort->getNodeDataIdFromName(splitVar[2]);
+    if(varId < 0) {
+        std::cout << "Error: No such variable: " << splitVar[2].c_str() << "\n";
+        size = -1;
+        return nullptr;
+    }
+
+    size = int(spCoreComponentSystem->getNumActuallyLoggedSamples());
+    double *data = new double[size];
+    std::vector< std::vector<double> > *pLogData = pPort->getLogDataVectorPtr();
+    for (size_t t=0; t<spCoreComponentSystem->getNumActuallyLoggedSamples(); ++t) {
+        data[t] = (*pLogData)[t][size_t(varId)];
+    }
+
+    return data;
+}
+
+int loadLibrary(const char *path)
+{
+    if(!gHopsanCore.loadExternalComponentLib(path)) {
+        printWaitingMessages(gHopsanCore, false, false);
+        return -1;
+    };
+    printWaitingMessages(gHopsanCore, false, false);
+    return 0;
+}
+
+int setStartTime(double value)
+{
+    startTime = value;
+    return 0;
+}
+
+int setTimeStep(double value)
+{
+    if(!spCoreComponentSystem) {
+        std::cout << "Error: No model is loaded!\n";
+        return -1;
+    }
+    spCoreComponentSystem->setDesiredTimestep(value);
+    return 0;
+}
+
+int setStopTime(double value)
+{
+    stopTime = value;
+    return 0;
+}
+
+int simulate()
+{
+    if(!spCoreComponentSystem) {
+        std::cout << "Error: No model is loaded!\n";
+        return -1;
+    }
+    std::cout << "Checking model... ";
+    if (spCoreComponentSystem->checkModelBeforeSimulation()) {
+        std::cout << "Success!\n";
+    }
+    else {
+        std::cout << "Failed!\n";
+        printWaitingMessages(gHopsanCore, false, false);
+        return -1;
+    }
+
+    std::cout << "Initializing model... ";
```

Comment:
I think that we have a preInitialize also, that must run

Reply:
It is automatically called from within `initialize`, so we can skip it here.